src/reactive-handler.ts

```diff
@@ -1,74 +1,39 @@
-import {
-    toString,
-    isUndefined,
-    unwrap,
-    ArrayConcat,
-    isArray,
-    ObjectDefineProperty,
-    getPrototypeOf,
-    isExtensible,
-    getOwnPropertyDescriptor,
-    getOwnPropertyNames,
-    getOwnPropertySymbols,
-    preventExtensions,
-    hasOwnProperty,
-} from './shared';
+import { toString, isArray, unwrap, isExtensible, preventExtensions, ObjectDefineProperty, freeze } from './shared';
+import { ReactiveMembrane } from './reactive-membrane';
+import { BaseProxyHandler, ReactiveMembraneShadowTarget } from './base-handler';
 
-import {
-    ReactiveMembrane,
-    ReactiveMembraneShadowTarget,
-    wrapDescriptor,
-} from './reactive-membrane';
+const getterMap = new WeakMap<() => any, () => any>();
+const setterMap = new WeakMap<(v: any) => void, (v: any) => void>();
 
-function wrapValue(membrane: ReactiveMembrane, value: any): any {
-    return membrane.valueIsObservable(value) ? membrane.getProxy(value) : value;
-}
-
-/**
- * Unwrap property descriptors will set value on original descriptor
- * We only need to unwrap if value is specified
- * @param descriptor external descrpitor provided to define new property on original value
- */
-function unwrapDescriptor(descriptor: PropertyDescriptor): PropertyDescriptor {
-    if (hasOwnProperty.call(descriptor, 'value')) {
-        descriptor.value = unwrap(descriptor.value);
+export class ReactiveProxyHandler extends BaseProxyHandler {
+    constructor(membrane: ReactiveMembrane, value: any) {
+        super(membrane, value);
```

Comment:
Do you even need a constructor if its just calling `super` with the same arguments?

Reply:
yeah, this can be removed all together!